### PR TITLE
format doc comments

### PR DESCRIPTION
**Pull request by danieleades**
this PR uses a slightly more comprehensive rustfmt config to

- reorder impl items to match trait definitions
- normalise and format doc comments
- merge and organise imports

### CODE REVIEW

File: src/de.rs [hunk 1]
```diff
@@ -32,8 +34,9 @@ where
     }
 
     fn read_padding_of<T>(&mut self) -> Result<()> {
-        // Calculate the required padding to align with 1-byte, 2-byte, 4-byte, 8-byte boundaries
-        // Instead of using the slow modulo operation '%', the faster bit-masking is used
+        // Calculate the required padding to align with 1-byte, 2-byte, 4-byte, 8-byte
+        // boundaries Instead of using the slow modulo operation '%', the faster
+        // bit-masking is used
```
Comment (hrektts) [inline src/de.rs:39]: What is the length of the comment you are trying to set? The default is [80](https://rust-lang.github.io/rustfmt/?search=&version=v1.5.2#comment_width).

Comment (danieleades) [reply src/de.rs:39]: i haven't set the line length, so whatever behaviour you see here is the default.

I used the following config-

```toml
version = "Two"
use_field_init_shorthand = true
imports_granularity="Crate"
wrap_comments = true
use_try_shorthand = true
format_code_in_doc_comments = true
format_strings = true
normalize_comments = true
normalize_doc_attributes = true
remove_nested_parens = true
reorder_impl_items = true
group_imports = "StdExternalCrate"
```

Comment (danieleades) [reply src/de.rs:39]: i haven't set the line length, so whatever behaviour you see here is the default.

I used the following config-

```toml
version = "Two"
use_field_init_shorthand = true
imports_granularity="Crate"
wrap_comments = true
use_try_shorthand = true
format_code_in_doc_comments = true
format_strings = true
normalize_comments = true
normalize_doc_attributes = true
remove_nested_parens = true
reorder_impl_items = true
group_imports = "StdExternalCrate"
```

Comment (hrektts) [reply src/de.rs:39]: Could you please use [default values](https://rust-lang.github.io/rustfmt/?version=v1.5.3&search=) for the config?
The fact that the last CI was executed successfully means that `rustfmt` had nothing to complain about.

Comment (danieleades) [reply src/de.rs:39]: this config is *consistent with* the default config in the sense that it makes additional changes that the default config ignores, but does not make any changes that cause errors when using the default config.

For example, [the default config ignores the max line width for doc comments entirely](https://rust-lang.github.io/rustfmt/?version=v1.5.2&search=#false%5C%20%5C(default%5C)%5C%3A)

> The fact that the last CI was executed successfully means that rustfmt had nothing to complain about.

introducing a stricter formatting than that applied with the default config is the point of this PR. If that's not of interest then this PR should just be closed

**Pull request merged**